epoch_test.go

```diff
@@ -24,6 +24,85 @@ import (
 	"go.uber.org/zap/zapcore"
 )
 
+func TestEpochHandleNotarizationFutureRound(t *testing.T) {
+	l := testutil.MakeLogger(t, 1)
+	bb := &testBlockBuilder{out: make(chan *testBlock, 2), in: make(chan *testBlock, 1)}
+
+	nodes := []NodeID{{1}, {2}, {3}, {4}}
+
+	// Create the two blocks ahead of time
+	blocks := createBlocks(t, nodes, bb, 2)
+
+	firstBlock := blocks[0].Block.(*testBlock)
+	secondBlock := blocks[1].Block.(*testBlock)
+
+	// re-create the block builder
+	bb = &testBlockBuilder{out: make(chan *testBlock, 1), in: make(chan *testBlock, 1)}
```

Comment:
would it be simpler to do something like this? 
```suggestion
	bb := &testBlockBuilder{}
	nodes := []NodeID{{1}, {2}, {3}, {4}}

	// Create the two blocks ahead of time
	blocks := createBlocks(t, nodes, bb, 2)

	firstBlock := blocks[0].Block.(*testBlock)
	secondBlock := blocks[1].Block.(*testBlock)

	bb.out = make(chan *testBlock, 1)
	bb.in = make(chan *testBlock, 1)
```

Reply:
Fixed